src/new/spawnmanager/spawnmanager.cpp

```diff
@@ -0,0 +1,953 @@
+                if (Owner->CurrentBurstIndex % 2 == 0)
+                    fire_coord = owner_ext->Fire_Coord(weapon_slot);
+                else
+                    fire_coord = owner_ext->Fire_Coord(weapon_slot, owner_type_ext->SecondSpawnOffset);
+
+                Coordinate spawn_coord = Coordinate(fire_coord.X, fire_coord.Y, fire_coord.Z + 10);
+
+                const auto rocket = RocketTypeClass::From_AircraftType(SpawnType);
+                if (rocket && rocket->IsCruiseMissile)
+                {
+                    spawn_coord.X -= 40;
+                    spawn_coord.Y -= 40;
+                }
+
+                /**
+                 *  Place the spawn in the world.
+                 */
+                DirStruct dir = Owner->PrimaryFacing.Current();
+                spawnee->Unlimbo(spawn_coord, dir.Get_Dir());
+
+                /**
+                 *  Cruise missiles spawn their takeoff animation.
+                 */
+                if (rocket && rocket->IsCruiseMissile && rocket->TakeoffAnim)
+                    new AnimClass(rocket->TakeoffAnim, spawnee->Coord, 2, 1, SHAPE_WIN_REL | SHAPE_CENTER, -10);
+
+                /**
+                 *  Reset burst since if we're done with this volley.
+                 */
+                if (i == SpawnControls.Count() - 1)
+                    Owner->CurrentBurstIndex = 0;
+
+                /**
+                 *  Missiles only take a destination once, so they go straight to the target.
+                 */
+                if (control->IsSpawnedMissile)
+                {
+                    Next_Target();
+                    spawnee->Assign_Destination(Target);
+                    spawnee->Assign_Mission(MISSION_MOVE);
+                }
+                /**
+                 *  Aircraft first "organize" next to the spawner.
+                 */
+                else
+                {
+                    CellClass* owner_cell = Owner->Get_Cell_Ptr();
+                    CellClass* adjacent_cell = &owner_cell->Adjacent_Cell(FACING_S);
+                    spawnee->Assign_Destination(adjacent_cell);
+                    spawnee->Assign_Mission(MISSION_MOVE);
+                }
+
+                break;
+            }
+
+            /**
+             *  The rocket is taking off (handled by the locomotor), so wait until it's done, then let it go.
+             */
+        case SpawnControlStatus::Takeoff:
+            {
+                if (control->ReloadTimer.Expired())
+                    Detach(spawnee);
+                break;
+            }
+
+            /**
+             *  The aircraft is preparing to attack.
+             */
+        case SpawnControlStatus::Preparing:
+            {
+                /**
+                 *  Missiles don't do this.
+                 */
+                if (control->IsSpawnedMissile)
+                    break;
+
+                /**
+                 *  If there's not target, return to base.
```

Comment:
```suggestion
                 *  If there's no target, return to base.
```

This comment seems a bit weird btw. The comparison below is `if (Target != nullptr)`. It seems more like "if there's a target, return to base"?